design/authenticators/authn_jwt/authn_jwt_solution_design.md

```diff
@@ -0,0 +1,465 @@
+   
+* simplest "mapping" option from JWT to host annotation
+   * Meet? Yes
+   * How? By implementing small interface
+   * Motivation? Will give us the flexibility to support any new future requirement and the maturity
+   to move it later to more "smarter" component to consume it from policy configuration
+
+* easily way to contribute code
+   * Meet? Yes
+   * How? By implementing new JWTConfiguration implementation class
+   * Motivation? Will give us the flexibility to support any new future requirement and it keeps the authenticator policy
+   simple to use
+     
+* support static keys in 1st phase
+   * Meet? Partially
+   * Why? At the time of writing, we have not yet begun the POC of how its works 
+   * How? The is interface with this 1 responsibility 
+
+* token-schema
+   * Meet? No
+   * Why? I suggest to delay this decision in order to keep a simple use and also have more flexibility 
+     by moving this logic to code
+     
+### Backwards compatibility
+None because this is a new feature.
+
+### Performance
+JWT authenticator performance should conform with our other authenticators with an average call time of 1 second.
+
+### Affected Components
+- Conjur
+
+- DAP
+
+## Open questions
+1. Do we want to support static keys in 1st phase?
+2. Which vendor constraints do we want to implement in 1st phase? for example, project-id is mandatory
+
+## Security
+We will validate we are are following JWT best practices 
+
+## Test Plan
+TODO
+
+## Delivery Plan
+
+### Total 1st milestone EE 54
+
+![authn-jwt-m1-task-dependencies](authn-jwt-m1-task-dependencies.png)
+
+#### Ramp-up EE 2
+* Learn authenticators code EE 2
+
+#### Designs EE 1
+* Design the status api call for JWT generic authenticator EE 1
+
+#### POC and Researches 3
+* Validate resource restrictions as part of load policy EE 3
+
+#### Implementations EE 20
+* Skeleton and empty classes according to HLD EE 3
+* Impl JWT token validation EE 2
+    * Class KeyAdapter
+    * Class ValidateJWTRestrictions
+* Impl JWT validate and decode EE 5
+    * Class JWTFetchClaimsToValidate
+    * Class GetIssValue
+    * Class JWTFetchSigningKey
+    * Class ValidateAndDecode
+* Impl JWT get identity EE 3
+    * Implement factory
+    * Implement interface 
+    * Implement both token & url
+* Refactor Validate restrictions EE 3
+* Integrate all parts EE 1
+* Enhance logs ans supportability EE 1
+* Impl status api according to api call EE 2
+
+#### Infrastructure EE 6
+* We have JWT token available as part of dev script EE 3
+* We have JWT token available as part of CI process EE 3
+
+
+#### Testing EE 13
+* Implement Conjur integration tests according to test plan EE 10
+* Implement Conjur integration tests according to test plan EE 3
+
+#### Security EE 5
+* POCs and mitigation according to JWT best practices EE 5
+
+#### Docs EE 3
+
+#### Demo EE 1
+
+
+### Total Post 1st milestone EE 5
+
+* Refactor all duplicates code EE 5
+
+### Total 1st milestone + Token schema 16
```

Comment:
not sure I understand - does token schema adds another 16SP?

Reply:
Yes, after we developed stage 1 we need another 16 SP to support token schema

Reply:
Can we sum the 1st development stage?
So it will say
developed stage 1 is XX SP + Token schema 17SP